### DISCUSSION TITLE
How to copy and paste shapes between instances on iPad?

### DISCUSSION THREAD

**Discussion by valiyaveedan** [excalidraw/excalidraw#3946] [Q&A] [answered]
How do I copy a shapes I drew and then paste in another  instance while on iPad? On the PC it works with Ctrl + C and Ctrl +V but when I get the context menu on iPad I am not able to see the copy and paste options. I have seen this in earlier versions. Was this removed?

**Comment by valiyaveedan** [comment]
This is working on android phones and tablets but not showing up on iPads.

**Comment by dwelle** [comment, marked as answer]
Hey, this is a valid concern. We've removed copy/paste on non-mobile devices because it was prone to bugs due to non-cross-browser support of reading from system clipboard, which is needed for the cases of copy/pasting across windows/apps (https://github.com/excalidraw/excalidraw/issues/2868).

That said, even if we re-enable it at least for devices like tablets, it likely won't work in your case due to Safari.

I've created an issue to track this https://github.com/excalidraw/excalidraw/issues/3949
